pkg/reactable/catalyze.go

```diff
@@ -0,0 +1,109 @@
+package reactable
+
+import (
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/event"
+)
+
+func (r *Reactable) tryAggravate(a *combat.AttackEvent) {
+	if a.Info.Durability < ZeroDur {
+		return
+	}
+
+	switch a.Info.Element {
+	case attributes.Electro:
+		if r.Durability[attributes.Quicken] < ZeroDur {
+			return
+		}
+		// Aggravate doesn't consume any gauge
+	default:
+		//Trigger must be Electro. Maybe add a panic/error?
+		return
+	}
+
+	//trigger event before attack is queued. this gives time for other actions to modify it
+	r.core.Events.Emit(event.OnAggravate, r.self, a)
+
+	//em isn't snapshot
+	em := r.core.Player.ByIndex(a.Info.ActorIndex).Stat(attributes.EM)
+	a.Info.FlatDmg += 1.15 * r.calcCatalyzeDmg(a.Info, em)
+	a.Info.Amped = true
+	a.Info.AmpType = combat.Aggravate
+}
+
+func (r *Reactable) trySpread(a *combat.AttackEvent) {
+	if a.Info.Durability < ZeroDur {
+		return
+	}
+
+	switch a.Info.Element {
```

Comment:
The other ones are not done to figure out the trigger is correct, but to figure out which element is incoming to apply the correct calcs. The default case is technically unreachable.

So in this case the switch on `a.Info.Element` is redundant and should be removed. However, you still need to check for `r.Durability[attributes.Quicken]`